static/js/click_handlers.js

```diff
@@ -763,6 +763,22 @@ $(function () {
         $(".settings-section" + sel + ", .settings-wrapper" + sel).addClass("show");
     });
 
+    // Deactivate stream
+    $("#settings_overlay_container").on("click", "#do_deactivate_stream_button", function () {
+        if ($("#deactivation_stream_modal .stream_name").text() !== $(".active_stream_row").find('.stream_name').text()) {
+            blueslip.error("Stream deactivation canceled due to non-matching fields.");
+            ui_report.message(i18n.t("Deactivation encountered an error. Please reload and try again."),
+               $("#home-error"), 'alert-error');
+        }
+        overlays.close_modal('deactivation_stream_modal');
+        $("#deactivation_stream_modal").remove();
+        $(".active_stream_row button").prop("disabled", true).text(i18n.t("Working…"));
+        var stream_name = $(".active_stream_row").find('.stream_name').text();
+        var stream_id = stream_data.get_sub(stream_name).stream_id;
+        var row = $(".active_stream_row");
+        settings_streams.delete_stream(stream_id, $("#organization-status"), row);
+    });
+
```

Comment:
I think this still belongs in `settings_streams.js`, just not inside the `.set_up`.